cmd/run/run.go

```diff
@@ -293,6 +293,18 @@ func DefaultConfig() *Config {
 	}
 }
 
+// TCPRandomPort tries to find a random TCP Port. If it can't find one, it panics. Else, it returns the port and a function that releases the port.
+// It is the responsibility of the caller to call the release function.
+func TCPRandomPort() (int, func()) {
```

Comment:
Sorry for being all over the place 😆 i was trying to understand how it works, and now I do

I think this function should call `Listen` to pick the port and then `Close` it immediately. So that when `http.ListenAndServe` runs, the port is available

Reply:
The part i don't understand is this: `net.Listen("tcp", "")` already starts listening on the port. So how come `http.ListenAndServer(port)`doesn't give a "address already in use"?

Reply:
Oh wait this is why we call the `defer release()` function...